docs/bricks_design.rst

```diff
@@ -0,0 +1,602 @@
+
+Bricks design explained
+=======================
+
+This is a runnable tutorial, which means you can copy its fragments 
+to an IPython notebook and run. The import section follows:
```

Comment:
I started writing it as IPython notebook, but it is not convenient to keep IPython notebook under version control. Outputs have to be deleted every time before commit and so on.